src/main/java/de/tum/in/www1/exerciseapp/domain/DragAndDropQuestionStatistic.java

```diff
@@ -93,7 +113,48 @@ public String toString() {
      */
     @Override
     public void addResult(SubmittedAnswer submittedAnswer, boolean rated) {
-        //TO-DO: Moritz Issig
+
+        if(submittedAnswer == null) {
+            return;
+        }
+
+        DragAndDropSubmittedAnswer ddSubmittedAnswer = (DragAndDropSubmittedAnswer)submittedAnswer;
+
+        if(rated) {
+            //increase the rated participants
+            setParticipantsRated(getParticipantsRated() + 1);
+
+            if(ddSubmittedAnswer.getMappings() != null) {
+                // increase rated dropLocationCounter if dropLocation is correct
+                for (DropLocationCounter dropLocationCounter: dropLocationCounters) {
+                    if(dropLocationCounter.getDropLocation().isDropLocationCorrect(ddSubmittedAnswer.getMappings())) {
+                        dropLocationCounter.setRatedCounter(dropLocationCounter.getRatedCounter() + 1);
+                    }
+                }
+            }
+            // increase rated correctCounter if answer is complete correct
+            if(getQuestion().isAnswerCorrect(ddSubmittedAnswer)) {
+                setRatedCorrectCounter(getRatedCorrectCounter() + 1);
+            }
+        }
+        // Result is unrated
+        else{
+            //increase the unrated participants
+            setParticipantsRated(getParticipantsUnrated() + 1);
```

Comment:
This line seems to be wrong as you are in the unrated branch but you call "setParticipantsRated(...)"